### ISSUE TITLE
Simple Solution For Database Tasks With Azure

### ISSUE THREAD

**Issue by metaskills** [rails-sqlserver/activerecord-sqlserver-adapter#442]
The following tasks are the most common and we currently rely on the user's giving us administrator level credentials to connect to SQL Server and perform said tasks via simple SQL statements provided by each task. 

* create_database
* drop_database

These tasks are much more complicated on Azure which has no `USE ?` notion. You have to re-connect, but it does have a master DB. Options to support these tasks include:

* Assume login is server/db admin. Connect with no DB bit (master) use TSQL https://msdn.microsoft.com/en-us/library/dn268335.aspx
* Delegate work to X platform Azure CLI https://github.com/Azure/azure-xplat-cli

**Comment by metaskills** [comment]
cc @TigerMint

**Comment by TigerMint** [comment]
hey @metaskills, Just to make sure we're on the same page, I am assuming we're talking about rake:db drop, create, etc. 

Probably okay to assume login is admin. If login/configs enable drop and create, then tasks succeed, if it fails, then login/configs needs correct permissions which is by design/security purposes. 

To use Azure X Plat CLI, it would require Azure Subscription Credentials, which is out of scope for this adapter.

**Comment by metaskills** [comment]
I agree on XPlat CLI being out of scope. It is also has some issues that I found while doing some research diligence. First, creating a server defaults to 2.0 version vs 12.0, see this issue https://github.com/Azure/azure-xplat-cli/issues/1837 That bug also meant that DBs created via the CLI would never be able to set a proper edition. Assuming the first was fixed, the second would be mood. But I really like the idea of assuming a SQL Server being created with admin rights and then using the adapter to create/drop via TSQL.
